src/NServiceBus.Core/IBusExtensions.cs

```diff
@@ -119,7 +119,7 @@ public static void Unsubscribe<T>(this IBus bus)
         {
             Guard.AgainstNull("bus", bus);
 
-            bus.Unsubscribe(typeof(T), new UnsubscribeOptions());
+            bus.UnsubscribeAsync(typeof(T), new UnsubscribeOptions());
```

Comment:
We need to fix this. The synchronous version should be obsoleted and only the async version should be exposed